### PR TITLE
Feature/ Create Admin

### PR DESCRIPTION
**Pull request by mmrshk**

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class SessionsController < ApplicationController
+  def new
```
Comment (mmrshk) [inline app/controllers/sessions_controller.rb:2]: All of this devise already has.

Comment (mmrshk) [reply app/controllers/sessions_controller.rb:2]: Read more about devise, please. And delete this controller.

Comment (mmrshk) [reply app/controllers/sessions_controller.rb:2]: Currently it's not your task, you can do it, but it's out of your scope.

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  private
+    # Use callbacks to share common setup or constraints between actions.
+    def set_user
+      @user = User.find(params[:id])
+    end
+
+    # Only allow a list of trusted parameters through.
+    def user_params
+      params.require(:user).permit(:name, :email, :tel, :role, :password, :password_confirmation)
+    end
```
Comment (mmrshk) [inline app/controllers/users_controller.rb:101]: extra left spacing

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        format.html { redirect_to users_url, notice: "User was successfully destroyed." }
+        format.json { head :no_content }
+      end
+    end
+  end
+
+  private
+    # Use callbacks to share common setup or constraints between actions.
+    def set_user
+      @user = User.find(params[:id])
```
Comment (mmrshk) [inline app/controllers/users_controller.rb:76]: 1. memoization
2. Why you used find not find_by?

File: app/helpers/admins_helper.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+module AdminsHelper
```
Comment (mmrshk) [inline app/helpers/admins_helper.rb:1]: Don't create files like this. Add files with methods.

File: app/helpers/sessions_helper.rb [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    session[:user_id] = user.id
+  end
+
+  def current_user
+    if session[:user_id]
+      @current_user ||= User.find_by(id: session[:user_id])
+    end
+  end
+
+  #Check if super_admin
```
Comment (mmrshk) [inline app/helpers/sessions_helper.rb:14]: redundant comments

File: test/controllers/sessions_controller_test.rb [hunk 6]
```diff
@@ -0,0 +1,19 @@
+require 'test_helper'
```
Comment (mmrshk) [inline test/controllers/sessions_controller_test.rb:1]: delete this

Comment (mmrshk) [inline test/controllers/sessions_controller_test.rb:1]: delete this

File: app/controllers/sessions_controller.rb [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  end
+
+  def create
+    user = User.find_by(email: params[:session][:email].downcase)
+    if (user && user.authenticate(params[:session][:password]))
+      log_in user
+      current_user
```
Comment (igorkasyanchuk) [inline app/controllers/sessions_controller.rb:9]: why you calling current_user

File: app/controllers/admins_controller.rb [hunk 8]
```diff
@@ -0,0 +1,15 @@
+class AdminsController < ApplicationController
+  before_action :current_user
```
Comment (igorkasyanchuk) [inline app/controllers/admins_controller.rb:2]: why do you call current_user here?

File: app/helpers/sessions_helper.rb [hunk 9]
```diff
@@ -0,0 +1,45 @@
+module SessionsHelper
+  
+  # Logs in the given user.
+  def log_in(user)
+    session[:user_id] = user.id
+  end
+
+  def current_user
```
Comment (igorkasyanchuk) [inline app/helpers/sessions_helper.rb:8]: devise already have this method of current_user

File: app/models/user.rb [hunk 10]
```diff
@@ -0,0 +1,21 @@
+class User < ApplicationRecord
+
+  before_save   :downcase_email
```
Comment (igorkasyanchuk) [inline app/models/user.rb:3]: move callbacks below validations, this is according to style guide

File: app/views/admins/users_show.html.erb [hunk 11]
```diff
@@ -0,0 +1,43 @@
+<% provide(:title, 'Admin/Users') %>
+<p id="notice"><%= notice %></p>
+
+<% if super_admin? %>
+  <% unless @users.nil? %>
```
Comment (igorkasyanchuk) [inline app/views/admins/users_show.html.erb:5]: replace unless with "if @users.present?"

File: app/views/layouts/application.html.erb [hunk 12]
```diff
@@ -10,6 +10,8 @@
   </head>
 
   <body>
+    <%= render 'layouts/header' %>
     <%= yield %>
+    <div><%= debug(params) if Rails.env.development? %></div>
```
Comment (igorkasyanchuk) [inline app/views/layouts/application.html.erb:15]: remove this debug, it's not needed really

File: app/views/users/_user.json.jbuilder [hunk 13]
```diff
@@ -0,0 +1,2 @@
+json.extract! user, :id, :name, :email, :created_at, :updated_at
```
Comment (igorkasyanchuk) [inline app/views/users/_user.json.jbuilder:1]: do you need this file?

File: app/views/users/show.html.erb [hunk 14]
```diff
@@ -0,0 +1,18 @@
+<% provide(:title, 'User profile') %>
+<p id="notice"><%= notice %></p>
```
Comment (igorkasyanchuk) [inline app/views/users/show.html.erb:2]: you see, it's evewhere, DRY the code

File: app/views/users/show.html.erb [hunk 15]
```diff
@@ -0,0 +1,18 @@
...
+
+<p>
+  <strong>Email:</strong>
+  <%= @user.email %>
+</p>
+
+<%= link_to 'Edit', edit_user_path(@user) %> |
+<%= link_to 'Back', users_path %>
+
+<p>Current user's id: <%= @current_user.id %></p>
```
Comment (igorkasyanchuk) [inline app/views/users/show.html.erb:17]: why you need @current_user instance variable
use devise, it has helper current_user

File: app/views/users/show.json.jbuilder [hunk 16]
```diff
@@ -0,0 +1 @@
+json.partial! "users/user", user: @user
```
Comment (igorkasyanchuk) [inline app/views/users/show.json.jbuilder:1]: do you need jbuilder files?

File: test/system/users_test.rb [hunk 17]
```diff
@@ -0,0 +1,45 @@
+require "application_system_test_case"
+
+class UsersTest < ApplicationSystemTestCase
```
Comment (igorkasyanchuk) [inline test/system/users_test.rb:3]: I suggest to use rspec also, because rspec is used on all projects

**Comment by manastyretskyi** [comment]
@stasfly Is this a joke? You haven't resolved any comments, no changes in about two weeks and you just forget about this PR all together and opened a new one?
CC @mmrshk do something about it!

**Comment by stasfly** [comment]
No, no I have "refurbish" this task and pull it into as an another branch (feature/admin-pages). This branch must be deleted, but I do not do it now because of some comments from coleagues. I'll cust it out soon.

**Comment by manastyretskyi** [comment]
If you close the PR comments will be present, you can't delete it completely.

**Pull request closed without merging**